h2o-core/src/main/java/water/api/RequestServer.java

```diff
@@ -314,12 +313,18 @@ public void doGeneric(String method, HttpServletRequest request, HttpServletResp
 
       resp.writeTo(response.getOutputStream());
 
+    } catch (IllegalArgumentException e) {  // This error type should be passed to clients.
```

Comment:
Just an idea... I believe that the `IllegalArgumentException` is quite general and there are many cases when this exception might be thrown. What if we catch an `Exception`, notifies user about this problem via `JettyHTTPD.sendResponseError(response, 500, e.getMessage());` and than we re-throw this exception. With this approach it will be logged, but not consumed. However it might expose some details about the backend, which might not be the desired behaviour, I am not familiar with the security concerns.